web/src/api/images/verify_app.ts

```diff
@@ -0,0 +1,278 @@
+      return;
+    }
+
+    if (req.method !== "POST") {
+      return errorNotAllowed(req.method!, res, req);
+    }
+
+    if (
+      !["reviewer", "admin"].includes(
+        req.body.session_variables["x-hasura-role"]
+      )
+    ) {
+      return errorHasuraQuery({
+        res,
+        req,
+        detail: "Unauthorized Role",
+        code: "unauthorized_role",
+      });
+    }
+
+    if (req.body.action?.name !== "verify_app") {
+      return errorHasuraQuery({
+        res,
+        req,
+        detail: "Invalid action.",
+        code: "invalid_action",
+      });
+    }
+
+    const {
+      app_id,
+      reviewer_name,
+      is_reviewer_app_store_approved,
+      is_reviewer_world_app_approved,
+    } = req.body.input;
+
+    if (
+      !app_id ||
+      !reviewer_name ||
+      typeof is_reviewer_app_store_approved !== "boolean" ||
+      typeof is_reviewer_world_app_approved !== "boolean"
+    ) {
+      return errorHasuraQuery({
+        res,
+        req,
+        detail:
+          "app_id, reviewer_name, is_reviewer_app_store_approved, is_reviewer_world_app_approved must be set.",
+        code: "required",
+      });
+    }
+
+    const reviewer_client = await getAPIReviewerGraphqlClient();
+    const { app: appMetadata } = await getAppMetadataSDK(
+      reviewer_client
+    ).GetAppMetadata({
+      app_id: app_id as string,
+    });
+
+    const app = appMetadata[0];
+    if (!app) {
+      return errorHasuraQuery({
+        res,
+        req,
+        detail: "App not found.",
+        code: "not_found",
+      });
+    }
+
+    const awaitingReviewAppMetadata = app.app_metadata.find(
+      (metadata) => metadata.verification_status === "awaiting_review"
+    );
+
+    if (!awaitingReviewAppMetadata) {
+      return errorHasuraQuery({
+        res,
+        req,
+        detail: "No app awaiting review.",
+        code: "invalid_verification_status",
+      });
+    }
+
+    const verifiedAppMetadata = app.app_metadata.find(
+      (metadata) => metadata.verification_status === "verified"
+    );
+
+    const s3Client = new S3Client({
+      region: process.env.ASSETS_S3_REGION,
+    });
+
+    if (!process.env.ASSETS_S3_BUCKET_NAME) {
+      throw new Error("AWS Bucket Name must be set.");
+    }
+
+    const bucketName = process.env.ASSETS_S3_BUCKET_NAME;
+    const sourcePrefix = `unverified/${app_id}/`;
+    const destinationPrefix = `verified/${app_id}/`;
+
+    // In case we have some stale images in the folder we iterate directly to delete
+    const listObjectsResponse = await s3Client.send(
+      new ListObjectsCommand({
+        Bucket: bucketName,
+        Prefix: destinationPrefix,
+      })
+    );
+
+    const verifiedImageKeysToDelete = listObjectsResponse.Contents?.map(
+      (object) => object.Key
+    );
+
+    if (verifiedImageKeysToDelete && verifiedImageKeysToDelete.length > 0) {
+      const expirePromises = verifiedImageKeysToDelete.map((key) =>
```

Comment:
Let's add a TODO comment here, this should be done async from a worker

Reply:
this whole flow should actually be async

Reply:
Added an issue on linear and todo